lib/spack/spack/fetch_strategy.py

```diff
@@ -1063,6 +1064,23 @@ def for_package_version(pkg, version):
     raise InvalidArgsError(pkg, version, **args)
 
 
+def from_url_scheme(url, *args, **kwargs):
```

Comment:
I noticed that the original code assumed all mirrors were `http(s)://` URLs, and it occurred to me that the actual fetch strategy that should be used would depend on the URLs `scheme`.  It wasn't clear to me where I might find a function for constructing a fetch strategy based on a URL's scheme, so I added one, here.